### PR TITLE
feat(regional): South Africa VAT Audit Report

### PR DESCRIPTION
**Pull request by Anuja-pawar**
- [x] Add two DocTypes- South Africa VAT Settings & South Africa VAT Accounts, 
        To fetch accounts for which VAT report should be generated

Screenshot 2021-07-14 at 12 24 30 PM

- [x] Create VAT Audit Report

Screenshot from 2021-08-09 11-22-58

- [x] Handle 0% tax rate (added custom field `Is Zero Rated` in Item)
- [x] Documentation

### CODE REVIEW

File: erpnext/regional/south_africa/setup.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
+# License: GNU General Public License v3. See license.txt
+
+from __future__ import unicode_literals
+
+# import frappe, os, json
+from frappe.permissions import add_permission, update_permission_property
+
+def setup():
```
Comment (deepeshgarg007) [inline erpnext/regional/south_africa/setup.py]:
```suggestion
def setup(company=None, patch=True):
```

File: erpnext/regional/report/vat_audit_report/vat_audit_report.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+			if self.invoices:
+				self.get_invoice_items(doctype)
+				self.get_items_based_on_tax_rate(doctype)
+				self.get_data(doctype)
+
+		return self.columns, self.data
+
+	def get_sa_vat_accounts(self):
+		self.sa_vat_accounts = frappe.get_list("South Africa VAT Account",
+			filters = {"parent":self.filters.company}, pluck="account")
```
Comment (deepeshgarg007) [inline erpnext/regional/report/vat_audit_report/vat_audit_report.py]:
```suggestion
			filters = {"parent": self.filters.company}, pluck="account")
```

File: erpnext/regional/report/vat_audit_report/vat_audit_report.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+				"label": "Account",
+				"fieldtype": "Link",
+				"options": "Account",
+				"width": 140
+			},
+			{
+				"fieldname": "invoice_number",
+				"label": "Reference",
+				"fieldtype": "Link",
+				"options": doctype,
```
Comment (deepeshgarg007) [inline erpnext/regional/report/vat_audit_report/vat_audit_report.py]: Passing Doctype won't work, all the columns even Purchase Invoice refers to Sales Invoice
Add a column for for document type and make invoice number a Dynamic Link
You can refer accounts receivable report for reference

File: erpnext/regional/report/vat_audit_report/vat_audit_report.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+		conditions = ""
+		for opts in (("company", " and company=%(company)s"),
+			("from_date", " and posting_date>=%(from_date)s"),
+			("to_date", " and posting_date<=%(to_date)s")):
+			if self.filters.get(opts[0]):
+				conditions += opts[1]
+
+		return conditions
+
+	def get_data(self,doctype):
```
Comment (deepeshgarg007) [inline erpnext/regional/report/vat_audit_report/vat_audit_report.py]:
```suggestion
	def get_data(self, doctype):
```

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]

File: erpnext/regional/report/vat_audit_report/vat_audit_report.py (hunk 3 above)
Comment (Anuja-pawar) [reply erpnext/regional/report/vat_audit_report/vat_audit_report.py]: Fixed

**Review by deepeshgarg007 — APPROVED** [review]

File: erpnext/regional/south_africa/setup.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+# License: GNU General Public License v3. See license.txt
+
+from __future__ import unicode_literals
+
+# import frappe, os, json
+from frappe.custom.doctype.custom_field.custom_field import create_custom_fields
+from frappe.permissions import add_permission, update_permission_property
+
+def setup(company=None, patch=True):
+	add_permissions()
```
Comment (nabinhait) [inline erpnext/regional/south_africa/setup.py:11]: Call make_custom_fields

File: erpnext/regional/south_africa/setup.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
+# License: GNU General Public License v3. See license.txt
+
+from __future__ import unicode_literals
+
+# import frappe, os, json
+from frappe.custom.doctype.custom_field.custom_field import create_custom_fields
+from frappe.permissions import add_permission, update_permission_property
+
+def setup(company=None, patch=True):
```
Comment (nabinhait) [inline erpnext/regional/south_africa/setup.py:10]: Write a patch to setup for existing South Africa accounts

File: erpnext/regional/report/vat_audit_report/vat_audit_report.json [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+ "modified_by": "Administrator",
+ "module": "Regional",
+ "name": "VAT Audit Report",
+ "owner": "Administrator",
+ "prepared_report": 0,
+ "ref_doctype": "GL Entry",
+ "report_name": "VAT Audit Report",
+ "report_type": "Script Report",
+ "roles": [
+  {
```
Comment (nabinhait) [inline erpnext/regional/report/vat_audit_report/vat_audit_report.json:22]: Remove roles from here and add via regional setup

**Comment by nabinhait** [comment]
Backport the PR to v13

**Comment by AfshanKhan** [comment]
@Anuja-pawar The backport command failed. You will have to backport it manually on V13

**Comment by Andriesvn** [comment]
Please Make this exact feature As it is here available to country = Namibia as well.

**Pull request merged**